pkg/streamingpromql/engine_test.go

```diff
@@ -1836,11 +1836,82 @@ func (t *timeoutTestingQueryTracker) Close() error {
 	return nil
 }
 
-func TestAnnotations(t *testing.T) {
```

Comment:
Is there a reason why you've split this test in half?

Reply:
I like shorter tests and it felt like a good place to split it up.

Reply:
I'm not entirely against it, but it is going to cause a bunch of merge conflicts with the upcoming Prometheus 3 changes (which introduce a bunch of new annotations) unless I rebase those.

Perhaps we can keep things as they were in this PR and then split the test in a later PR?

Reply:
I think this will create conflicts regardless. Lets see where we are at when either this or the Prometheus 3 PR's are close to merging.